code/src/components/Main.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import Header from './Header';
+import ThoughtList from './ThoughtList';
+
+
+const HAPPY_THOUGHTS_URL = 'https://happy-thoughts-technigo.herokuapp.com/thoughts';
+const Main = () => {
+    const [thoughts, setThoughts] = useState([]);
+    const [loading, setLoading] = useState(true)
+
+    const fetchList = () => {
+        setLoading(true)
+        fetch(HAPPY_THOUGHTS_URL)
+            .then((res) => res.json())
+            .then((data) => setThoughts(data))
+            .catch((error) => console.error(error))
+            .finally(() => setLoading(false))
+    }
+    const handleLikes = (thoughtId, event) => {
```

Comment:
I notice that when you like a post in the list the page refreshes instead of just adding the like. Maybe you could try: 
`  const handleLikes = (thoughtId) => {
    const options = {
      method: 'POST',
    };
   const uri = `${HAPPY_THOUGHTS_URL}/${thoughtId}/like`;
    fetch(uri, options)
      .then((res) => res.json())
      .then((data) => {
        fetchList();
      });
  };`

See more on the button further down.